File: src/test/java/com/dotnomi/fabricdependencyinjection/testclasses/postconstruct/multiple/MultiplePostConstructService.java
package com.dotnomi.fabricdependencyinjection.testclasses.postconstruct.multiple;

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct;

@ModScoped
@SuppressWarnings("unused")
public class MultiplePostConstructService {
  private boolean firstBoolean;
  private boolean secondBoolean;

  @PostConstruct
  private void firstInitialize() {
    this.firstBoolean = true;
  }

  @PostConstruct
  private void secondInitialize() {
    this.secondBoolean = true;
  }

  public boolean isFirstBoolean() {
    return this.firstBoolean;
  }

  public boolean isSecondBoolean() {
    return this.secondBoolean;
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/postconstruct/multiple/MultiplePostConstructService.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.postconstruct.multiple

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct

@ModScoped
class MultiplePostConstructService {
    var isFirstBoolean: Boolean = false
        private set
    var isSecondBoolean: Boolean = false
        private set

    @PostConstruct
    private fun firstInitialize() {
        this.isFirstBoolean = true
    }

    @PostConstruct
    private fun secondInitialize() {
        this.isSecondBoolean = true
    }
}
